### PR TITLE
Add some comments to the functions and variables

### PR DESCRIPTION
**Pull request by seeforschauer**

### CODE REVIEW

**Comment by w84thesun** [comment]
Does this linter checks unexported functions and methods?

**Comment by AlekSi** [comment]
It checks all code. If you mean if it checks _documentation_ for unexpected methods and functions, then no, and currently, there is no such linter, according to Gophers Slack.

**Comment by AlekSi** [comment]
Yeah, this PR does not refs 587

**Comment by seeforschauer** [comment]
> Yeah, this PR does not refs #587

Removed #587. 

But it detects, isn't it?

**Comment by seeforschauer** [comment]
> > Removed #587.
> 
> Both title and description still reference this issue.
> 
> > But it detects, isn't it?
> 
> Sorry, I don't understand the question. It (what it?) detects what?

It detects that comment absent for a variable or a function.

**Comment by AlekSi** [comment]
Yes, but only for exported names. #587 is about all functions/methods, including unexpected ones.

**Comment by seeforschauer** [comment]
> Yes, but only for exported names. #587 is about all functions/methods, including unexpected ones.

understood, thank you.

**Comment by seeforschauer** [comment]
> Both title and description still reference this issue.

Renamed PR. Actually, it is just to make smaller PRs, here is not only comments that were added.

File: internal/handlers/common/error.go [hunk 1]
```diff
@@ -34,19 +34,46 @@ const (
...
-	ErrProjectionInEx    = ErrorCode(31253) // Location31253
-	ErrProjectionExIn    = ErrorCode(31254) // Location31254
-	ErrRegexOptions      = ErrorCode(51075) // Location51075
+	// ErrBadValue for wrong input.
+	ErrBadValue = ErrorCode(2) // BadValue
+
+	// ErrFailedToParse failed to parse user input.
+	ErrFailedToParse = ErrorCode(9) // FailedToParse
+
+	// ErrTypeMismatch for $sort, when it is expected field sort to be of type object but is has other type.
```
Comment (rumyantseva) [inline internal/handlers/common/error.go:43]: Are those codes some MongoDB's codes or something FerretDB-specific? The `it is expected field sort to be` part of the comment doesn't look clear to me. Is _"field sort"_ actually _field's sort_ or is it a field called _"sort"_? I was trying to find something Mongo-related for it in MongoDB docs, but I'm not sure where to look for it.

File: internal/handlers/pg/pgdb/placeholder.go [hunk 2]
```diff
@@ -16,8 +16,10 @@ package pgdb
 
 import "strconv"
 
+// Placeholder is a numbered mark where to put variable in the postgresql query.
```
[the remark this answers is not in the document]
Comment (seeforschauer) [reply internal/handlers/pg/pgdb/placeholder.go:19]: both are obvious :D

[the remark this answers is not in the document]
Comment (seeforschauer) [reply internal/handlers/pg/pgdb/placeholder.go:19]: they are obvious to us, may I put here both definitions? WDYT?

[the remark this answers is not in the document]
Comment (seeforschauer) [reply internal/handlers/pg/pgdb/placeholder.go]: changed both comments (here and below)

File: internal/handlers/common/error.go [hunk 3]
```diff
@@ -34,20 +34,46 @@ const (
...
-	ErrInvalidArg        = ErrorCode(28667) // Location28667
-	ErrSliceFirstArg     = ErrorCode(28724) // Location28724
-	ErrProjectionInEx    = ErrorCode(31253) // Location31253
-	ErrProjectionExIn    = ErrorCode(31254) // Location31254
-	ErrRegexOptions      = ErrorCode(51075) // Location51075
+	// ErrBadValue for wrong input.
+	ErrBadValue = ErrorCode(2) // BadValue
+
+	// ErrFailedToParse failed to parse user input.
+	ErrFailedToParse = ErrorCode(9) // FailedToParse
```
Comment (rumyantseva) [inline internal/handlers/common/error.go:41]: Overall, here and below I miss verbs in comments. Usually, in the Go source code, there is a verb explaining what the variable is used for. For example, for errors, it could be "indicates" [like in the os package](https://cs.opensource.google/go/go/+/refs/tags/go1.18.2:src/os/error.go;l=18).

On the other hand, in the Go source code, not all the errors are documented as detailed as you did it here :) So, please treat my comment as an opinion, not a call for action.

Comment (seeforschauer) [reply internal/handlers/common/error.go:41]: I welcome your feedback. I believe you added an example here to be more understandable to read when I add this. Anyway, this is a pure PR for comments.

File: internal/clientconn/conn.go [hunk 4]
```diff
@@ -79,10 +79,6 @@ type newConnOpts struct {
 
 // newConn creates a new client connection for given net.Conn.
 func newConn(opts *newConnOpts) (*conn, error) {
-	if opts.handler == nil {
```
Comment (rumyantseva) [inline internal/clientconn/conn.go:82]: Why is this deleted? Is this `opts.handler == nil` condition checked somewhere else?

File: internal/handlers/pg/pgdb/pool.go [hunk 5]
```diff
@@ -41,7 +41,10 @@ const (
 )
 
 var (
-	ErrNotExist     = fmt.Errorf("schema or table does not exist")
+	// ErrNotExist is an error when schema or table do not exist whily attempt to access it.
```
Comment (rumyantseva) [inline internal/handlers/pg/pgdb/pool.go:44]:
```suggestion
	// ErrNotExist is an error when schema or table does not exist while attempting to access it.
```

File: internal/handlers/pg/pgdb/pool.go [hunk 6]
```diff
@@ -41,7 +41,10 @@ const (
...
+	ErrNotExist = fmt.Errorf("schema or table does not exist")
+
+	// ErrAlreadyExist is an error when schema or table already exist while creating it.
```
Comment (rumyantseva) [inline internal/handlers/pg/pgdb/pool.go:47]:
```suggestion
	// ErrAlreadyExist is an error when schema or table already exists while creating it.
```

File: internal/util/version/version.go [hunk 7]
```diff
@@ -40,6 +40,7 @@ var (
 	branch string
 )
 
+// Info contains this build details.
```
Comment (rumyantseva) [inline internal/util/version/version.go:43]:
```suggestion
// Info contains current build details.
```

File: internal/util/version/version.go [hunk 8]
```diff
@@ -51,6 +52,7 @@ type Info struct {
 
 var info *Info
 
+// Get returns info.
```
Comment (rumyantseva) [inline internal/util/version/version.go:55]:
```suggestion
// Get returns current build's info.
```

**Review by rumyantseva — CHANGES_REQUESTED** [review]
Added a couple of questions/suggestions, and a couple of typo corrections.
In general, looks good to me.

File: internal/clientconn/conn.go (hunk 4 above)
Comment (seeforschauer) [reply internal/clientconn/conn.go:82]: it's deleted bcz of linter error (cannot compare handler to nil)

Comment (AlekSi) [reply internal/clientconn/conn.go:82]: Well, `opts.handler` is an interface, and it can be nil. Please revert that change

Comment (rumyantseva) [reply internal/clientconn/conn.go:82]: @AlekSi should it panic though? Or should it return an error instead?

Comment (seeforschauer) [reply internal/clientconn/conn.go:82]: > Well, `opts.handler` is an interface, and it can be nil. Please revert that change

done

Comment (AlekSi) [reply internal/clientconn/conn.go:82]: > should it panic though? Or should it return an error instead?

It should not be possible for a user to do something that would trigger that case, so that's a programmer's error, and we "handle" it by panicking

Comment (rumyantseva) [reply internal/clientconn/conn.go:82]: Ok! Sounds reasonable!

**Review by rumyantseva — APPROVED** [review]
LGTM

File: internal/handlers/common/error.go [hunk 9]
```diff
@@ -25,7 +25,7 @@ import (
 
 //go:generate ../../../bin/stringer -linecomment -type ErrorCode
 
-// ErrorCode represents wire protocol error code.
+// ErrorCode represents FerretDB error codes.
```
Comment (AlekSi) [inline internal/handlers/common/error.go:28]: I think it makes it less clear. Everything in the FerretDB code base is about FerretDB. And the value of that type represents a single error code, not many error codes.

File: internal/handlers/pg/pgdb/placeholder.go [hunk 10]
```diff
@@ -16,8 +16,10 @@ package pgdb
 
 import "strconv"
 
+// Placeholder stores the number of the relevant placeholder of the query.
 type Placeholder int
 
+// Next increases the identifier value for the next variable in the postgresql query.
```
Comment (AlekSi) [inline internal/handlers/pg/pgdb/placeholder.go:22]:
```suggestion
// Next increases the identifier value for the next variable in the PostgreSQL query.
```

File: internal/types/binary.go [hunk 11]
```diff
@@ -20,14 +20,29 @@ package types
 type BinarySubtype byte
 
 const (
-	BinaryGeneric    = BinarySubtype(0x00) // generic
-	BinaryFunction   = BinarySubtype(0x01) // function
+	// BinaryGeneric BSON generic binary subtype.
```
Comment (AlekSi) [inline internal/types/binary.go:23]: There should be a verb like "represents"

File: internal/types/types.go [hunk 12]
```diff
@@ -48,6 +48,7 @@ import (
 	"time"
 )
 
+// MaxDocumentLen is maximum BSON object size.
```
Comment (AlekSi) [inline internal/types/types.go:51]:
```suggestion
// MaxDocumentLen is the maximum BSON object size.
```

I think it is worth checking comments with Grammarly

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/clientconn/conn.go [hunk 13]
```diff
@@ -82,7 +82,6 @@ func newConn(opts *newConnOpts) (*conn, error) {
 	if opts.handler == nil {
 		panic("handler required")
 	}
-
```
[the remark this answers is not in the document]
Comment (seeforschauer) [reply internal/clientconn/conn.go]: reverted.

File: internal/types/binary.go (hunk 11 above)
Comment (seeforschauer) [reply internal/types/binary.go]: added.

File: internal/types/regex.go [hunk 14]
```diff
@@ -21,17 +21,39 @@ import (
 )
 
 var (
+	// ErrOptionNotImplemented for unimplemented regex option.
```
[the remark this answers is not in the document]
Comment (seeforschauer) [reply internal/types/regex.go]: added verbs.

File: internal/handlers/common/error.go (hunk 9 above)
Comment (seeforschauer) [reply internal/handlers/common/error.go]: reverted.

**Review by rumyantseva — DISMISSED** [review]
I don't have major questions about the comments. This is a significant improvement in documentation already!

**Comment by seeforschauer** [comment]
> There are merge conflicts

resolved

**Review by rumyantseva — DISMISSED** [review]
LGTM

File: internal/types/regex.go [hunk 15]
```diff
@@ -21,17 +21,39 @@ import (
...
+	ErrTrailingBackslash = fmt.Errorf("Regular expression is invalid: \\ at end of pattern")
+
+	// ErrNothingToRepeat indicates invalid regex: nothing to repeat.
+	ErrNothingToRepeat = fmt.Errorf("Regular expression is invalid: nothing to repeat")
+
+	// ErrInvalidClassRange indicates that range out of order in character class.
+	ErrInvalidClassRange = fmt.Errorf("Regular expression is invalid: range out of order in character class")
+
+	//nolint:godot // false positive
+	// ErrUnsupportedPerlOp indicates for unrecognized character after (? or (?-.
```
Comment (w84thesun) [inline internal/types/regex.go:52]: This could be rephrased:
```suggestion
	// ErrUnsupportedPerlOp indicates unrecognized character after the grouping sequence start.
```

**Review by w84thesun — CHANGES_REQUESTED** [review]

**Review by w84thesun — APPROVED** [review]

**Review by rumyantseva — DISMISSED** [review]
LGTM

File: internal/util/testutil/testutil.go [hunk 16]
```diff
@@ -20,6 +20,7 @@ import (
 	"testing"
 )
 
+// Ctx returns a new empty context.
```
Comment (AlekSi) [inline internal/util/testutil/testutil.go:23]:
```suggestion
// Ctx returns test context.
```

In the future, it may return a context with a deadline/timeout for graceful cancelation, etc.

**Review by AlekSi — APPROVED** [review]
:shipit:
